dll/win32/comctl32/propsheet.c

```diff
@@ -3580,6 +3580,27 @@ PROPSHEET_DialogProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
         }
         SetFocus(GetDlgItem(hwnd, IDOK));
       }
+#ifdef __REACTOS__
+      { /* try to fit it into the desktop */
+          RECT rcWork;
+          RECT rcDlg;
+          int dx, dy;
+          int cx, cy;
+
+          GetWindowRect(hwnd, &rcDlg);
+          SystemParametersInfo(SPI_GETWORKAREA, 0, &rcWork, 0);
+          dx = rcDlg.right - rcWork.right;
+          dy = rcDlg.bottom - rcWork.bottom;
+          cx = rcDlg.right - rcDlg.left;
+          cy = rcDlg.bottom - rcDlg.top;
+          if (dx > 0)
```

Comment:
It would be safer to check `rcDlg.right > rcWork.right` here, otherwise this is subject to integer overflow and might yield strange results.

Reply:
This was done just before our code block, in the call  `if (PROPSHEET_SizeMismatch (hwnd, psInfo))` or `PROPSHEET_AdjustSizeWizard (hwnd, psInfo);`

Reply:
Hmm, let me try to be clearer:
The `dx` value being checked here may be the result of a calculation that experienced an integer overflow. Extreme example: `rcDlg.right=0x80000000` (smallest negative number), `rcWork.right=0x7fffffff` (largest positive number). Clearly the dialog should fit (the right side of the work area is (much) larger than the right side of the dialog). But `dx` would be `1` in this case, meaning you'd try to move the window.

If instead you directly check `if (rcDlg.right > rcWork.right)`, you will do it before any overflow occurs and therefore get the right result.
That said, you can still get an overflow in that case (`0x7fffffff - (-1 aka 0xffffffff) = 0x80000000`) so maybe it's not worth worrying about.